python_modules/dagster/dagster/_core/definitions/definitions_class.py

```diff
@@ -96,6 +96,20 @@ def created_repo():
 
         self._created_repo = created_repo
 
+    @public
+    def get_job_def(self, name: str) -> JobDefinition:
+        """Get a job definition by name. If you passed in a an UnresolvedAssetJobDefinition
+        (return value of define_asset_job) it will be resolved to a JobDefinition when returned
+        from this function."""
+
+        check.str_param(name, "name")
+        return self.get_resolved_repo().get_job(name)
+
+    def get_resolved_repo(self) -> RepositoryDefinition:
```

Comment:
Might we want to cache this?

Reply:
`self._created_repo = self._created_repo.compute_repository_definition()`

Something else than this on line 110?

Reply:
Oops, I read this too fast, this looks right. It strikes me as a little weird to overwrite the attribute, but seems fine.